src/scala/org/pantsbuild/zinc/compiler/OutputUtils.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2012 Typesafe, Inc. <http://www.typesafe.com>
+ */
+
+package org.pantsbuild.zinc.compiler
+
+import java.io.File
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{FileVisitResult, Files, Path, Paths, SimpleFileVisitor}
+import java.util.jar.{JarEntry, JarInputStream, JarOutputStream}
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+object OutputUtils {
+
+  /**
+   * Sort the contents of the `dir` in lexicographic order.
+   *
+   * @param dir File handle containing the contents to sort
+   * @return sorted set of all paths within the `dir`
+   */
+  def sort(dir:File): mutable.TreeSet[(Path, Boolean)] = {
+    val sorted = new mutable.TreeSet[(Path, Boolean)]()
+
+    val fileSortVisitor = new SimpleFileVisitor[Path]() {
+      override def preVisitDirectory(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        sorted.add(path, false)
```

Comment:
Eek. Sorry. Just looked at the javadocs for this: https://docs.oracle.com/javase/7/docs/api/java/util/zip/ZipEntry.html#isDirectory()

So the boolean isn't really necessary as long as you append a `/` here, and then later check whether it ends with a slash. Sorry!

Reply:
(...and the slash is required in order to not confuse consumers of the jar.)